src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -407,12 +408,17 @@ calculatePrincipalExchangePaymentClaims paymentScheduleAligned issuerPaysLeg cur
 -- https://www.fpml.org/spec/fpml-5-11-3-lcwd-1/html/confirmation/schemaDocumentation/schemas/fpml-shared-5-11_xsd/complexTypes/FloatingRateCalculation/finalRateRounding.html
 roundRate : Decimal -> Rounding -> Decimal
 roundRate rate rounding =
-  fromBigNumeric $ BN.round rounding.precision roundingMode (fromNumeric rate)
-    where
-      roundingMode = case rounding.roundingDirection of
-        Up -> RoundingCeiling
-        Down -> RoundingFloor
-        Nearest -> RoundingHalfUp
+  let exponent : Int = truncate $ logBase 10.0 rate
```

Comment:
I feel that this function should be exposed on one of the main packages (e.g. `Daml.Finance.Util`) and tested separately from the Fpml swap

Reply:
I think it looks good for positive numbers. Probably won't work for negative numbers due to log of a negative number. In that case maybe flip the sign before rounding, and then flip it back again. The logic for what is `up` and `down` probably would have to be flipped as well.

I was thinking whether this could even go into the `Daml` repo. If `BN.round` is indeed gone, seems something is missing from the main library that could be of interest to others.

Reply:
I tried modifying your solution to add support for negative numbers, including adding a couple of tests for negative numbers:
https://github.com/digital-asset/daml-finance/pull/1208
Feel free to take anything from there that you think would be useful.

Reply:
I think the rounding logic is quite general. As long as we use built in Daml functions (like `BN.round` or whatever comes next) it should be fine here, but if we have to write our own rounding function I think it should be outside so that it can be easily used by others, either in `Daml.Finance.Util` or in the Daml repo. 

Now that Remy is writing new "official" rounding functions for the Daml repo, our function can probably stay here for the time being, and then we update it to the official ones (and delete ours) when available. What do you think?

Reply:
Makes sense to me, I would open an issue to track this item and revisit the rounding logic (maybe we can use a new milestone for 3.x changes)